pyrival/graphs/maximum_matching.py

```diff
@@ -0,0 +1,66 @@
+import itertools
+from random import randint
+
+
+_DEFAULT_PRIME = 1073750017
+
+
+def maximum_matching(edges, mod=_DEFAULT_PRIME):
+    """
+    Returns the maximum cardinality matching of an undirected general graph
+    Uses a randomized algorithm to compute the rank of the Tutte matrix
+    The rank of the Tutte matrix is equal to twice the size of the maximum matching with high probability
+
+    Complexity: O(n ^ 3) worst case, O(n * m) with high constant factor on average
+
+    :param edges: a list of edges, assume nodes can be anything numbered from 0 to max number in edges
+    :param mod: a large random prime
+    :return: the maximum cardinality matching of the graph
+    """
+
+    edges = [(u, v) for u, v in edges if u != v]
+
+    if not edges:
+        return 0
+
+    n = max(itertools.chain(*edges)) + 1
+    matrix = _get_tutte_matrix(n, edges, mod)
+    return _gauss(matrix, mod) // 2
+
+
+def _get_tutte_matrix(n, edges, mod):
+    matrix = [[0] * n for _ in range(n)]
+
+    for u, v in edges:
+        u, v = min(u, v), max(u, v)
+        val = randint(1, mod - 1)
+        matrix[u][v], matrix[v][u] = val, mod - val
```

Comment:
@bjorn-martinsson you're right, it can be simplified and made clearer. And about using `v` twice, that was unintentional. Thanks for spotting it.

Reply:
Even u, v = min(u, v), max(u, v) is unnecessary since the order of u and v doesn't matter.